src/chainstate/stacks/mod.rs

```diff
@@ -360,7 +360,9 @@ impl Txid {
     }
 }
 
-/// How a transaction may be appended to the Stacks blockchain
+/// Used to determine
+/// - how a transaction may be appended to the Stacks blockchain
+/// - how a transaction will be validated for the the post transaction endpoint
```

Comment:
nit: "the the post transaction"? is there a noun missing between the two "the's"

Reply:
why are you reusing this enum?

In your `openapi.yaml` description you say

```
            If mempool_admission_check=="unconfirmed", the transaction will only be validated against unconfirmed state.
            If mempool_admission_check=="confirmed", the transaction will only be validated against confirmed state.
            If unspecified, the transaction will be accepted if it validates against either unconfirmed or confirmed state.
```

It seems kind of confusing to me that we went from confirmed/unconfirmed to onchain/offchain.

Is that like too newbie of a confusion?

Reply:
They are pretty much synonyms, although I do understand your confusion here. 
I think we don't typically use the phrase OffChainOnly/OnChainOnly very often, but it's something we learn with exposure.

Reply:
I actually think it would be better to use a separate enum for this (or even just a bool would suffice, since you have only two options).  This enum is specifically meant to represent the way we encode the anchor mode for transactions.  It's conceptually unclear what the `::Any` option means, for example.